seed.mts

```diff
@@ -0,0 +1,197 @@
+];
+const possibleFeminineProducts = ["Tampons", "Pads", "Incontinence Pads"];
+const possiblePets = ["Cat", "Dog"];
+const possibleOtherItems = [
+    "Garlic",
+    "Ginger",
+    "Chillies",
+    "Spices",
+    "Hot Water Bottles",
+    "Blankets",
+];
+
+const eventNamesWithNumberData = ["Shipping Labels Downloaded"];
+
+const eventNamesWithNoData = [
+    "No Status",
+    "Request Denied",
+    "Pending More Info",
+    "Called and Confirmed",
+    "Called and No Response",
+    "Ready to Dispatch",
+    "Received by Centre",
+    "Collection Failed",
+    "Parcel Collected",
+    "Delivered",
+    "Delivery Failed",
+    "Delivery Cancelled",
+    "Fulfilled with Trussell Trust",
+    "Shopping List Downloaded",
+    "Out for Delivery",
+    "Request Deleted",
+];
+
+const earliestDate = new Date(2024, 0, 1); // 2024/01/01
+const latestDate = new Date(2025, 0, 1); // 2025/01/01
+
+async function generateSeed(): Proimse<void> {
+    const seed = await createSeedClient({
+        dryRun: process.env.DRY !== "0",
+    });
+
+    await seed.$resetDatabase(); // Clears all existing data in the database, but keep the structure
+
+    await seed.clients((generate) =>
+        generate(500, {
+            fullName: (ctx) => copycat.fullName(ctx.seed),
+            phoneNumber: (ctx) => copycat.phoneNumber(ctx.seed),
+            address1: (ctx) => copycat.streetAddress(ctx.seed),
+            address2: (ctx) => copycat.streetAddress(ctx.seed),
+            addressTown: (ctx) => copycat.city(ctx.seed),
+            addressCounty: (ctx) => copycat.state(ctx.seed),
+            addressPostcode: (ctx) => copycat.countryCode(ctx.seed),
+            deliveryInstructions: (ctx) => copycat.sentence(ctx.seed, { maxWords: 20 }),
+            familyId: (ctx) => copycat.uuid(ctx.seed),
+            dietaryRequirements: (ctx) =>
+                copycat.someOf(
+                    ctx.seed,
+                    [0, possibleDietaryRequirements.length],
+                    possibleDietaryRequirements
+                ),
+            feminineProducts: (ctx) =>
+                copycat.someOf(
+                    ctx.seed,
+                    [0, possibleFeminineProducts.length],
+                    possibleFeminineProducts
+                ),
+            petFood: (ctx) => copycat.someOf(ctx.seed, [0, possiblePets.length], possiblePets),
+            otherItems: (ctx) =>
+                copycat.someOf(ctx.seed, [0, possibleOtherItems.length], possibleOtherItems),
+            extraInformation: (ctx) => copycat.sentence(ctx.seed, { maxWords: 20 }),
+            flaggedForAttention: (ctx) => copycat.bool(ctx.seed),
+            signpostingCallRequired: (ctx) => copycat.bool(ctx.seed),
+        })
+    );
+
+    await seed.collectionCentres([
+        {
+            name: "Brixton Hill - Methodist Church",
+            acronym: "BH-MC",
+        },
+        {
+            name: "Clapham - St Stephens Church",
+            acronym: "CLP-SC",
+        },
```

Comment:
Could you also please put "Delivery"/"DLVR" as a collection centre?

It's really not ideal (see https://softwiretech.atlassian.net/browse/VFB-55) that delivery is represented like this, because it's the most common real-world situation.

For test data, it would be great if about half of the parcels were Delivery and the other half were one of the (other) collection centres - is that possible?

Reply:
Added delivery as a collection centre now. Choosing delivery 50% of the time turned out to be non-trivial. The current implementation should pick a collection centre randomly for each parcel (33% chance) so I'll leave this as is for now.

Reply:
For context, this is how the parcels page looks currently
![image](https://github.com/VauxhallFoodbank/vauxhall-foodbank/assets/114006382/aa638ab4-d7aa-4ced-8f81-e71695b704ee)

Reply:
(The postcode could do with some more refining - do pick from a set of 10 of postcodes, some of which would be in the congestion charge list? But the new starters are itching to start on the project so I'm keen to get this merged first and do some refining at a later point)

Reply:
This is looking great!
I understand about wanting to get things going.

I think your suggestion about randomly picking from a predefined list of postcodes is good, especially as the client will want to see that the Congestion Charge functionality works as they expect; also so the local devs don't forget that CC functionality exists.

Will it be easy to have everyone replace their data once that's done?

Reply:
While I was waiting, I thought I might as well implement this now:
![image](https://github.com/VauxhallFoodbank/vauxhall-foodbank/assets/114006382/fc82dc3f-62a2-4417-b39c-778cda6cd85d)

Reply:
Superhero!